src/clipboard-copy-element.js

```diff
@@ -13,7 +13,9 @@ function copy(button: HTMLElement) {
 }
 
 function copyTarget(button: Element, id: string) {
-  const content = button.ownerDocument.getElementById(id)
+  // $FlowFixMe until https://github.com/facebook/flow/pull/7181 is approved
+  const root = 'getRootNode' in Element.prototype ? button.getRootNode() : button.ownerDocument
```

Comment:
Hey @jorgecasar thanks for the PR!

Should this be `getRootNode()` or `getRootNode({ composed: true })`? I can see use cases for both.

Reply:
According with the specifications of [Node.getRootNode](https://developer.mozilla.org/en-US/docs/Web/API/Node/getRootNode):

> composed: A Boolean that indicates whether the shadow root should be returned (`false`, the default), or a root node beyond shadow root (`true`).

it should be `false` because we want to find the `id` inside the Shadow Dom where the button is on. If we put `composed: true` the `document` will be returned and we were in the same situation as `ownerDocument` and we can't find the `id` inside the Shadow Dom of the web components.

Reply:
What if the element to copy is outside of the shadow root though?

Reply:
It couldn't be supported. There are 2 DOMs, global DOM (Document) or shadow DOM. Then the element should be in one of those. Using Shadow DOM is common to be in the same DOM.

Otherwise we have to rethink the component to provide the element it self to be copied.